test/de/schildbach/pte/live/TpgProviderLiveTest.java

```diff
@@ -0,0 +1,115 @@
+
+/**
+ * @author Andreas Schildbach
+ */
+public class TpgProviderLiveTest extends AbstractProviderLiveTest {
+    public TpgProviderLiveTest() {
+        super(new TpgProvider(secretProperty("tpg.api_authorization")));
+    }
+
+    @Test
+    public void nearbyStationsByCoordinate() throws Exception {
+        final NearbyLocationsResult result = queryNearbyStations(Location.coord(46209460, 6167140)); // Parc et plage des Eaux-Vives
+        print(result);
+    }
+
+    @Test
+    public void queryDepartures() throws Exception {
+        final QueryDeparturesResult result = queryDepartures("8501238", false); // Genève-Paquis
+        print(result);
+    }
+
+    @Test
+    public void queryDeparturesSuburbanTrain() throws Exception {
+        final QueryDeparturesResult result = queryDepartures("8516155", false); // Lancy-Pont-Rouge
+        print(result);
+    }
+
+    @Test
+    public void queryDeparturesTram() throws Exception {
+        final QueryDeparturesResult result = queryDepartures("8587907", true); // Plainpalais
+        print(result);
+    }
+
+    // @Test
+    // public void queryDeparturesTrolley() throws Exception {
+    //     final QueryDeparturesResult result = queryDepartures("8591177", false); // Hardplatz
+    //     print(result);
+    // }
+
+    @Test
+    public void queryDeparturesInvalidStation() throws Exception {
+        final QueryDeparturesResult result = queryDepartures("999999", false);
+        assertEquals(QueryDeparturesResult.Status.INVALID_STATION, result.status);
+    }
+
+    @Test
+    public void suggestLocations() throws Exception {
+        final SuggestLocationsResult result = suggestLocations("Champel");
+        print(result);
+    }
+
+    @Test
+    public void shortTrip() throws Exception {
+        final Location from = new Location(LocationType.STATION, "8592913", "Genève", "Trembley");
+        final Location to = new Location(LocationType.STATION, "8595027", "Cointrin", "De-Joinville");
+        final QueryTripsResult result = queryTrips(from, null, to, new Date(), true, null);
+        print(result);
+        final QueryTripsResult laterResult = queryMoreTrips(result.context, true);
+        print(laterResult);
+    }
+
+    // @Test
+    // public void trip() throws Exception {
+    //     // final Location from = new Location(LocationType.STATION, "8593059", Point.from1E6(46163584, 6106258), "Plan-les-Ouates",
+    //     //         "Plan-les-Ouates, Galaise");
+    //     final Location from = new Location(LocationType.STATION, "8592913", "Genève", "Trembley");
+    //     final Location to = new Location(LocationType.STATION, "8593217", Point.from1E6(46325307, 6064940), "Cessy",
+    //             "Cessy-Les Hauts");
+    //     final QueryTripsResult result = queryTrips(from, null, to, new Date(), true, null);
+    //     print(result);
+    //     final QueryTripsResult laterResult = queryMoreTrips(result.context, true);
+    //     print(laterResult);
+    // }
```

Comment:
I'm not able to make it works because with this type of Location ctor, it returns "UNKNOWN LOCATION".

Reply:
I usually use the `suggestLocations()` method / test to find locations that should be valid in the region of the provider. Did you try that?